add user namespace to use for the subjectAccessReview in oauthproxy

JameelB opened the pull request:
## Description
Add a user namespace parameter to the sydnesis template to be used for the subjectAccessReview in oauthproxy to allow the user who provisioned the service authorization to the Fuse dashboard.

phantomjinx committed:
fix(server): Notifies of stale integration after connection update

 When a connection is modified, checks are performed comparing the new
 connection against the cached versions inside any draft integrations and
 deployed integrations. If an integration is found to not be equivalent,
 ie. its metadata is out-of-date, then a message is logged accordingly.

 * app/common/model...
  * Adds default equivalence methods to model interfaces that test only
    selected properties of instances. Thus, weaker than equality but does
    not produce irritating false-positives.

 * app/common/model.../ConnectionBase
 * app/common/model.../Connector
 * app/common/model.../Extension
  * Adds the Version.Auxiliary annotation to certain methods. This stops
    them from being added, by Immutables library, to hashcode and equals
    methods. The properties behind these methods are only populated on
    augmentation by the REST API hence are not critical to the object and
    including them means they always fail equality checks with the version
    of the same object fetched from the DataManager.

 * IntegrationUpdateHandler
  * On update of a connection, compares the connection to both draft and
    deployed integrations.

 * IntegrationUpdateHandlerTest
  * Provides testing for use-cases when a connection may be modified

Fixes #1032

TovaCohen committed:
user doc: Initial draft of user doc for ServiceNow connector

TovaCohen committed:
user doc: Change "pages not supported" to "pagination not supported"

zregvart committed:
feat(server): model changes for multi-flow inte...

...grations

Adds the notion of Flow and having a list of flows within an
Integration. Properties that relate to single Camel route now reside
within the flow.

TovaCohen committed:
user doc: Adds doc about using Apicurio to edit OpenAPI spec.

zregvart committed:
refactor(server): followup on the JSON detection

This is a followup on the JSON detection via regex in #3373 raised in
review (on IRC).

TovaCohen committed:
user doc: Redo Concur doc to downplay implementation/production site info

TovaCohen committed:
Merge pull request #3435 from TovaCohen/FUSEDOC-2380-concur

user doc: Redo Concur doc to downplay implementation/production site info

oscerd committed:
fix(slack connector): Slack Connector: when used with username as action the action will fail

pure-bot[bot] committed:
Merge pull request #3438 from oscerd/synd-3437

zregvart committed:
feat(ui): changes for multi-flow integrations

This changes the model to match the server multi-flow integration model,
and adds the ability to switch between two flows when editing an
integration by specifying the flow id as the fragment in the URL.

oscerd committed:
feature(slack-connector): Add consumer action

pure-bot[bot] committed:
Merge pull request #3439 from oscerd/slack-consumer-demo

JameelB committed:
add a sar_project parameter to be used for the subjecAccessReview in oauthproxy

zregvart committed:
Merge pull request #3425 from zregvart/issue/3373-followup

refactor(server): followup on the JSON detection

oscerd committed:
refactor(slack-connector): improved the Verifier

pure-bot[bot] committed:
Merge pull request #3442 from oscerd/slack-verifier-improve

nicolaferraro committed:
Fix #3444: making telegram tech preview

pure-bot[bot] committed:
Merge pull request #3445 from nicolaferraro/3444-master

TovaCohen committed:
user doc: Limit Per Page limitations in this release, import set config requirement

TovaCohen committed:
user doc: Merge modifications

TovaCohen committed:
Merge pull request #3366 from TovaCohen/FUSEDOC-2632-servicenow

user doc: user doc for ServiceNow connector

rhuss committed:
feature(upgrade): Adapte templates and operator for referencing a registry (#3448)

Now the operator can be called with a `-registry <registry>` argument for using that registry when instantiation the upgrade pod

A productised Dockerfile should use this option to point the target registry.

Also, the template generator has an options `--upgrade-tag` to directly specify the upgrade tag.

Fixes #3440

zregvart committed:
chore(upgrade): migration scripts for 1.4.x (#3447)

* chore(upgrade): migration scripts for 1.4.x

This adds migration scripts for OpenShift resources from version 1.3.x
to 1.4.x.

Fixes #3443.

TovaCohen committed:
user doc: Mark Concur as tech preview. Add "webhook" to main title for that content.

TovaCohen committed:
Merge pull request #3452 from TovaCohen/FUSEDOC-2380-concur

user doc: Mark Concur as tech preview.

TovaCohen committed:
Merge pull request #3413 from TovaCohen/FUSEDOC-2633-apicurio

user doc: Adds doc about using Apicurio to edit OpenAPI spec.

TovaCohen committed:
user doc: Say what Concur is, offset param not required, implementation cred update.

TovaCohen committed:
Merge pull request #3455 from TovaCohen/FUSEDOC-2380-concur

user doc: Say what Concur is, offset param not required.

david-martin commented on install/operator/pkg/syndesis/configuration/configuration.go:
```diff
@@ -162,11 +165,31 @@ func openShiftConsoleUrlFromEnv(config map[string]string, syndesis *v1alpha1.Syn
 }
 
+func userNamespaceFromEnv(config map[string]string, syndesis *v1alpha1.Syndesis) {
+	if v, ok := getString(config, EnvUserNamespace); ok {
```
@JameelB to keeps backwards compatibility, and have a sensible default, I think the `USER_NAMESPACE` should default to the value of the `OPENSHIFT_PROJECT` env var.

JameelB replied:
@david-martin That's being handled here: https://github.com/JameelB/syndesis/pull/1/files#diff-8f63b1158cd0c8188207de68b0013e69R183

david-martin replied:
perfect

elvisisking committed:
Fix for Issue 3459 apicurio - warnings are not updated when changed via gui and saved
- the problem was that the CustomConnectorRequest instance is being reused and all the validation properties were not being cleared before re-validation.
- now clearing out these properties before requesting validation: actionsSummary, errors, and warnings.

lburgazzoli committed:
feat(api provider): integration support

- multi flow #3468
- rest dsl #3469

oscerd committed:
refactor(kafka-connector): Minor stuff

pure-bot[bot] committed:
Merge pull request #3494 from oscerd/kafka-connector-minor-fixes

lburgazzoli committed:
feat(api provider): integration support

elvisisking committed:
Proposed fix for Issue 2717 Provide more information about the imported Swagger specification
- Added a new section on the review page for the API information (includes API name and description).
- i18n'd the ApiConnectorReviewComponent

nicolaferraro committed:
Initial refactoring of connector-generator

mcoker committed:
bug(blank-slate): update blank slate primary action button

gashcrumb committed:
fix: avoid NPE and hide basic filter if there's no output data shape in any previous connection; fixes #3506

mcoker committed:
bug(connections): fix "configuration required" spacing in connection list

pure-bot[bot] committed:
Merge pull request #3509 from michael-coker/issue-3508

lburgazzoli committed:
Convert old-style integrations into flows (backward compatibility) #3472

pure-bot[bot] committed:
Merge pull request #3510 from gashcrumb/issue-3506

pure-bot[bot] committed:
Merge pull request #3501 from phantomjinx/issue-1032

pure-bot[bot] committed:
Merge pull request #3507 from elvisisking/2717

pure-bot[bot] committed:
Merge pull request #3511 from michael-coker/issue-3489

pure-bot[bot] committed:
Merge pull request #3483 from elvisisking/3459

igarashitm committed:
chore: Update to AtlasMap 1.36.0

Contains:
* Data Preview with final design (atlasmap/atlasmap#451)
* Support receiving FieldActions metadata from outside (atlasmap/atlasmap#483)
* Create GenerateFieldActionsMojo (atlasmap/atlasmap#482)
* bug fixes
  * atlasmap/atlasmap#544
  * atlasmap/atlasmap#495
  * atlasmap/atlasmap#375
  * atlasmap/atlasmap#515
  * atlasmap/atlasmap#521
  * atlasmap/atlasmap#491

pure-bot[bot] committed:
Merge pull request #3518 from igarashitm/atlasmap-1.36.0

gashcrumb committed:
feat: add the flow ID to the route instead of using the fragment

nicolaferraro committed:
Fix #3515: do not override command from operator image

pure-bot[bot] committed:
Merge pull request #3522 from nicolaferraro/3515-master

oscerd committed:
feature(slack-connector): Add Metadata to retrieve channels in the consuming action

pure-bot[bot] committed:
Merge pull request #3525 from oscerd/slack-consuming-meta

lburgazzoli committed:
feat(api provider): integration support

oscerd committed:
refactor(slack-connector): refactoring of readResponse methods

oscerd committed:
refactor(connector): Minor typos fix

pure-bot[bot] committed:
Merge pull request #3527 from oscerd/slack-consuming-meta

pure-bot[bot] committed:
Merge pull request #3528 from oscerd/typos-kafka-slack

lburgazzoli committed:
Create a Camel endpoint to use as end of API integrations #3524

lburgazzoli committed:
Fix stale integration detection after switching to flows

pure-bot[bot] committed:
Merge pull request #3530 from syndesisio/flows-final

nicolaferraro committed:
First api provider generator

oscerd committed:
refactor(slack-connector): Webhook and token must be Raw and secret

pure-bot[bot] committed:
Merge pull request #3535 from oscerd/minor-slack

nicolaferraro committed:
Linked API generator with Syndesis runtime

nicolaferraro committed:
Adding return code logic

rhuss committed:
Allow multiple issue templates (#3537)

nicolaferraro committed:
Adding validation endpoint

nicolaferraro committed:
Adding validation endpoint

nicolaferraro committed:
Fixing PMD errors and tests

nicolaferraro committed:
Shrink integration size by using a external icon

nicolaferraro committed:
Fix license

pure-bot[bot] committed:
Merge pull request #3544 from nicolaferraro/flows-master

rhuss committed:
Update simple issue template to remove boilerplate
